tchannel/tornado/peer.py

```diff
@@ -656,6 +677,34 @@ def _update_heap(self, peer):
         peer.rank = rank
         self.peer_heap.update_peer(peer)
 
+    def get_for_request(self, hostport):
```

Comment:
Consider calling this `get_isolated` or something similar and make it private while you're at it.